action/recipes/config_test.go

```diff
@@ -0,0 +1,223 @@
+			wantErr: nil,
+			opts:    Config{},
+		},
+		{
+			name:    "empty coreboot opts",
+			wantErr: nil,
+			opts: Config{
+				Coreboot: map[string]CorebootOpts{},
+			},
+		},
+		{
+			name: "missing required coreboot opts",
+			opts: Config{
+				Coreboot: map[string]CorebootOpts{
+					"coreboot-A": {},
+				},
+			},
+			wantErr: ErrRequiredOptionUndefined,
+		},
+		{
+			name:    "required coreboot opts present",
+			wantErr: nil,
+			opts: Config{
+				Coreboot: map[string]CorebootOpts{
+					"coreboot-A": {
+						Common: commonDummy,
+					},
+				},
+			},
+		},
+		{
+			name:    "required coreboot opts present 2",
+			wantErr: nil,
+			opts: Config{
+				Coreboot: map[string]CorebootOpts{
+					"coreboot-A": {
+						Common:   commonDummy,
+						Specific: CorebootSpecific{},
+					},
+				},
+			},
+		},
+		{
+			name:    "missing common opts",
+			wantErr: ErrRequiredOptionUndefined,
+			opts: Config{
+				Coreboot: map[string]CorebootOpts{
+					"coreboot-A": {
+						Common:   CommonOpts{},
+						Specific: CorebootSpecific{},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := ValidateConfig(tc.opts)
+			assert.ErrorIs(t, err, tc.wantErr)
+		})
+	}
+}
+
+func TestConfigReadAndWrite(t *testing.T) {
+	configOriginal := Config{
+		Coreboot: map[string]CorebootOpts{},
+		Linux:    map[string]LinuxOpts{},
+		Edk2:     map[string]Edk2Opts{},
+	}
+
+	tmpDir := t.TempDir()
+	configFilepath := filepath.Join(tmpDir, "test.json")
+
+	// Write
+	err := WriteConfig(configFilepath, configOriginal)
+	assert.NoError(t, err)
+
+	// Read
+	configNew, err := ReadConfig(configFilepath)
+	assert.NoError(t, err)
+
+	// Compare
+	equal := cmp.Equal(configOriginal, configNew)
+	if !equal {
+		fmt.Println(cmp.Diff(configOriginal, configNew))
+		assert.True(t, equal, "written and read configuration are not equal")
+	}
+}
+
+func TestConfigEnvVars(t *testing.T) {
+	commonDummy := CommonOpts{
+		SdkURL:        "ghcr.io/9elements/firmware-action/coreboot_4.19:main",
+		Arch:          "dummy",
+		RepoPath:      "dummy/dir/",
+		DefconfigPath: "dummy",
+		OutputDir:     "dummy/dir/",
+	}
+
+	testCases := []struct {
+		name             string
+		wantErr          error
+		url              string
+		urlExpected      string
+		repoPath         string
+		repoPathExpected string
+		envVars          map[string]string
+	}{
+		{
+			name:             "no env vars",
+			wantErr:          nil,
+			url:              commonDummy.SdkURL,
+			urlExpected:      commonDummy.SdkURL,
+			repoPath:         commonDummy.RepoPath,
+			repoPathExpected: commonDummy.RepoPath,
+			envVars:          map[string]string{},
+		},
+		{
+			name:             "env vars",
+			wantErr:          nil,
+			url:              "ghcr.io/$TEST_ENV_VAR/coreboot_4.19:main",
+			urlExpected:      commonDummy.SdkURL,
+			repoPath:         commonDummy.RepoPath,
+			repoPathExpected: commonDummy.RepoPath,
+			envVars: map[string]string{
```

Comment:
Happened in a few places in that test file alone, you could either change it leave it as CI doesn't complain it's fine by me.

Reply:
This is result of `task format`, which in turn is `gofumpt -w action`

Reply:
`gofumpt` by default does that. I think it might be possible to reconfigure `gofumpt`, but I do not want to mess with it.

As such, I will leave it as is.